### PR TITLE
WIP Test xarray release candidate

### LINKED ISSUE #224: Test against xarray release candidate
Xarray has a [v0.10 release candidate](https://github.com/pydata/xarray/releases/tag/v0.10.0rc1) out.  It's a big release.  We should check that nothing breaks on our end, both for our own sake and as a service to xarray in the case we find any bugs.

More generally moving forward, I think it would be useful to have a Travis branch that tests against the xarray dev branch; it can be permitted to have failures.  Xarray does the same for pandas and dask; e.g. [here](https://github.com/pydata/xarray/blob/v0.10.0rc1/ci/requirements-py36-pandas-dev.yml) for the pandas dev branch and the ["allow failures"](https://github.com/pydata/xarray/blob/v0.10.0rc1/.travis.yml#L46) section of their `.travis.yml`.

@micahkim23, I'm going to put you on this one, as I think it's a good entree into the packaging/release/CI side of things.  For the v0.10rc1 part, I would clone that release to your local machine, make sure its tests pass, and then make sure our tests pass using it.  For the dev branch/CI part, just mimic the examples I gave above.

Let us know if you have any questions or trouble.

### PR DESCRIPTION
**Pull request by micahkim23**
Closes #224

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,16 +6,21 @@ notifications:
...
-      env: CONDA_ENV=py27
-    - python: 2.7
-      env: CONDA_ENV=py27-min
-    - python: 3.4
-      env: CONDA_ENV=py34
-    - python: 3.5
-      env: CONDA_ENV=py35
-    - python: 3.6
-      env: CONDA_ENV=py36
+  - python: 2.7
```
Comment (spencerahill) [inline .travis.yml:9]: Please revert the indentation to what it was before.  Unlike Python, YAML doesn't care much about whitespace, so this is just stylistic.  But I like the original indentation we had, and -- more importantly -- it's consistent with the indentation in the rest of the fall and in all of our other .yml files.

File: docs/whats-new.rst [hunk 2]
```diff
@@ -28,6 +28,9 @@ Corrected link to documentation badge on repository main page (:pull:213). By Da
 Enhancements
 ~~~~~~~~~~~~
 
+- Create Travis branch that tests against the xarray dev branch. This
```
Comment (spencerahill) [inline docs/whats-new.rst]: "Create Travis CI environment that tests against the xarray development branch."  Omit second sentence.  Move from Enhancements to new section "Testing"

**Review by spencerahill — CHANGES_REQUESTED** [review]
~~A few minor cleanup points.~~ One minor cleanup point.  Again the AppVeyor failure is something unrelated.

Start your next commit message with `[ci-skip]` so that it doesn't trigger the CI; these changes won't need it.

File: .travis.yml (hunk 1 above)
Comment (micahkim23) [reply .travis.yml:9]: the old indentation in the first commit throws an error `syntax error: (<unknown>): did not find expected '-' indicator while parsing a block collection at line 9 column 5` in the travis build. I tested the syntax with http://lint.travis-ci.org/.

Comment (spencerahill) [reply .travis.yml:9]: Sorry, you're totally right!  Should have checked the Travis log from your first commit before acting like I knew what I was talking about.

So ignore that comment.

**Review by spencerahill — APPROVED** [review]

**Comment by spencerahill** [comment]
Thanks @micahkim23 !

File: .travis.yml (hunk 1 above)
Comment (spencerkclark) [reply .travis.yml:9]: For what it's worth, the following checks out on http://lint.travis-ci.org/ (note I've de-indented `allow_failures:` by two spaces), but it's too late now :)

```
language: python
sudo: false
notifications:
  email: false

matrix:
  fast_finish: true
  include:
    - python: 2.7
      env: CONDA_ENV=py27
    - python: 2.7
      env: CONDA_ENV=py27-min
    - python: 3.4
      env: CONDA_ENV=py34
    - python: 3.5
      env: CONDA_ENV=py35
    - python: 3.6
      env: CONDA_ENV=py36
    - python: 3.6
      env: CONDA_ENV=py36-xarray-dev
  allow_failures:
    - python: 3.6
      env: CONDA_ENV=py36-xarray-dev

before_install:
  - if [[ "$TRAVIS_PYTHON_VERSION" == "2.7" ]]; then
      wget https://repo.continuum.io/miniconda/Miniconda2-latest-Linux-x86_64.sh -O miniconda.sh;
    else
      wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh;
    fi
  - bash miniconda.sh -b -p $HOME/miniconda
  - export PATH="$HOME/miniconda/bin:$PATH"
  - hash -r
  - conda config --set always_yes yes --set changeps1 no
  - conda update -q conda
  - conda info -a

install:
  - conda env create --file ci/environment-$CONDA_ENV.yml
  - source activate test_env
  - pip install -e .

script:
  - py.test aospy --cov=aospy --cov-report term-missing
after_success:
  - coveralls
```

Comment (spencerahill) [reply .travis.yml:9]: Ah good catch!  I prefer this.  Will switch to this at some point.

**Pull request merged**